sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -168,6 +168,28 @@ static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 	return 0;
 }
 
+#define HDA_ANA_CODEC_AUTOSUSPEND_DELAY_MS 1000
+
+static void skl_hda_set_ana_codec_autosuspend(struct snd_soc_card *card)
+{
+	struct snd_soc_dai *codec_dai;
+	struct hdac_hda_priv *hda_pvt;
+	struct hdac_device *hdev;
+
+	codec_dai = snd_soc_card_get_codec_dai(card, "Analog Codec DAI");
```

Comment:
how certain are we that the name will never change? If we have to rely on it maybe better replace it with a macro?

Reply:
OK, will add a macro.
thx.

Reply:
@lyakh very certain :) this is defined in hdac_hda.c -- but it's true if this would ever change, it could silently break jack detection in this machine driver. So maybe a more future-proof solution is to apply the delay just to all codecs. This is what the legacy driver does as well. How about it @jason77-wang ?

Reply:
@kv2019i  it is not easy to set for all codecs, I listed all codec-dais, they are:
intel-hdmi-hifi1
intel-hdmi-hifi2
intel-hdmi-hifi3
Analog Codec DAI
Digital Codec DAI
dmic-hifi
dmic-hifi
snd-soc-dummy-dai
We only need to set the delay for intel-hdmi-hifi1 and Analog Codec DAI, if skip other codec-dais, we also need to compare with name string.

So I add two macros which will be used by hdac_hda.c and skl_hda_dsp_generic.c, please take a look at my new patch.